server/src/lib.rs

```diff
@@ -129,7 +132,14 @@ pub async fn run_server(config: Config) -> Result<()> {
             };
             let service = DbFactoryService::new(db_factory.clone());
             if let Some(addr) = config.rpc_server_addr {
-                tokio::spawn(run_rpc_server(addr, db_factory, logger_clone));
+                tokio::spawn(run_rpc_server(
+                    addr,
+                    config.rpc_server_tls,
+                    config.rpc_server_cert.clone(),
+                    config.rpc_server_key.clone(),
```

Comment:
(2) and then here you could just pass `config.rpc_server_tls.then((config.rpc_server_cert.unwrap(), config.rpc_server_key.unwrap()))`

Reply:
and let's at least print a warning if somebody passes a cert without a key, or cert and key without enabling tls, etc.

Unless there's a magic `clap` option to enforce that if tls is set, then cert and key are required, but I'm not aware of any